tests/test_s3.py

```diff
@@ -1656,6 +1656,243 @@ def test_bucket_replication_policy_disable(self):
+        client = session.client("s3")
+
+        p = self.load_policy(
+            {
+                "name": "CheckForPublicBlocks-Absent",
+                "resource": "s3",
+                "filters": [
+                    {"Name": bname},
+                    {
+                        "type": "check-public-block",
+                        "scope": "Any",
+                        "state": "absent"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "set-public-block"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+
+        # Test that there was a bucket with missing public blocks
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        # Verbose test to make sure that the public blocks are enabled on the buckets
+        try:
+            response = client.get_public_access_block(
+                Bucket=bname)['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                response = None
+            else:
+                raise
+        if response:
+            for key in response.keys():
+                self.assertEqual(response[key], True)
+
+    def test_set_public_block_disable_all(self):
+        bname = 'mypublicblock'
+
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(s3, "S3_AUGMENT_TABLE", [])
+
+        session_factory = self.replay_flight_data("test_s3_public_block_disable_all")
+        session = session_factory()
+        client = session.client("s3")
+
+        p = self.load_policy(
+            {
+                "name": "CheckForPublicBlocks-Present",
+                "resource": "s3",
+                "filters": [
+                    {"Name": bname},
+                    {
+                        "type": "check-public-block",
+                        "scope": "Any",
+                        "state": "present"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "set-public-block",
+                        "state": "disable"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+
+        # Test that there was a bucket with missing public blocks
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        # Verbose test to make sure that the public blocks are enabled on the buckets
+        try:
+            response = client.get_public_access_block(
+                Bucket=bname)['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                self.assertEqual(response, None)
+            else:
+                raise
+        if response:
```

Comment:
this whole block feels like copy/paste from before and is very confusing, if you disable, then check that directly.  if your recording and it takes time, aka eventual consistency, use
```
if self.recording:
    time.sleep(n)
```
for delays

Reply:
Agreed, I'll keep in mind this use case for sleeping on recording. It was mostly copy pasta from the s3 resource where I need to check if the response is not None. Thanks for pointing this out, I have updated in d680d31.